File: IOS/appPickerRegex/appPickerRegex/ViewController.m
//
//  ViewController.m
//  appPickerRegex
//
//  Created by Jorge on 21/04/24.
//

#import "ViewController.h"

@interface ViewController ()
{
    NSArray *expresiones;
    NSArray *equivalencia;
}
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    equivalencia = @[ @"Número base 10", @"Hexadecimal", @"Binario", @"Número base 8", @"Número telefónico" ];
    
    _listaRegex = [ [ NSArray alloc ] initWithObjects:[ NSRegularExpression regularExpressionWithPattern:@"^[0-9]+$" options:0 error:NULL ], [ NSRegularExpression regularExpressionWithPattern:@"^[0-9A-Fa-f]+$" options:0 error:NULL ], [ NSRegularExpression regularExpressionWithPattern:@"^[0-1]+$" options:0 error:NULL  ], [ NSRegularExpression regularExpressionWithPattern:@"^[0-7]+$" options:0 error:NULL ], [ NSRegularExpression regularExpressionWithPattern:@"^\\(\\d{3}\\)\\-\\d{3}\\-\\d{4}$" options:0 error:NULL ], nil ];

    
    self.pickerExpresiones.delegate = self;
    self.pickerExpresiones.dataSource = self;
    
    _etiquetaResultado.text = @" ... ";
}

-(NSInteger)numberOfComponentsInPickerView:(UIPickerView *)pickerView {
    
    return 1;
    
}

- (NSInteger)pickerView:(UIPickerView *)pickerView numberOfRowsInComponent:(NSInteger)component {
    
    return [equivalencia count];
    
}

- (NSString *)pickerView:(UIPickerView *)pickerView titleForRow:(NSInteger)row forComponent:(NSInteger)component {
    return equivalencia[row];
}

- (void)pickerView:(UIPickerView *)pickerView didSelectRow:(NSInteger)row inComponent:(NSInteger)component {
    NSRegularExpression *expresionEvaluar = [ _listaRegex objectAtIndex:row ];
    
    NSString *texto = _textoRevisar.text;
    
    int numberOfMatches = (int)[expresionEvaluar numberOfMatchesInString:texto options:0 range:NSMakeRange(0, [texto length])];
    
    if (numberOfMatches > 0) {
        
        _etiquetaResultado.text = @"Número válido";
        
    } else {
        
        _etiquetaResultado.text = @"Número no válido";
        
    }
    
}

@end

File: IOS/SWIFT_IOS/appPickerViewProductos/appPickerViewProductos/ViewController.swift
//
//  ViewController.swift
//  appPickerViewProductos
//
//  Created by Jorge Cisneros on 27/05/24.
//

import UIKit

class ViewController: UIViewController, UIPickerViewDelegate, UIPickerViewDataSource {
    
    //--------------------------------------------------------
    
    func loadImage(named name: String) -> UIImage? {
        let filePath = Bundle.main.path(forResource: name, ofType: nil)
        if let path = filePath {
            let image = UIImage(contentsOfFile: path)
            return image
        }
        return nil
    }
    
    //--------------------------------------------------------
    
    let carro_color = [ [ "VW", "Toyota", "Ford", "Chrysler", "Honda", "Mazda" ], ["🚗", "🚌", "🚘"] ]
    
    let imagenes = ["bici1.jpg","camioneta1.jpg", "carro1.jpg", "BICI_1_copy.png","BICI_2.png", "carro11.png"]
    
    var images: [UIImage] = []
    
    @IBOutlet weak var label1: UILabel!
    @IBOutlet weak var texto1: UITextField!
    @IBOutlet weak var pickerView1: UIPickerView!
    @IBOutlet weak var imageView1: UIImageView!
    
    //--------------------------------------------------------
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        pickerView1.delegate = self
        pickerView1.dataSource = self
        
        for imageName in imagenes {
            if let image = loadImage(named: imageName) {
                images.append(image)
            }
        }
        
    }
    
    //--------------------------------------------------------
    
    func numberOfComponents(in pickerView: UIPickerView) -> Int {
        return 2
    }
    
    //--------------------------------------------------------
    
    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
        return carro_color[component].count
    }
    
    //--------------------------------------------------------
    
    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
        return carro_color[component][row]
    }
    
    //--------------------------------------------------------
    
    func pickerView(_ pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
        let carro = carro_color[0][pickerView.selectedRow(inComponent: 0)]
        
        let color = carro_color[1][pickerView.selectedRow(inComponent: 1)]
        
        let imagen = images[pickerView.selectedRow(inComponent: 0)]
        
        imageView1.image = imagen
        
        texto1.text = carro + " " + color
        
        label1.text = carro + " " + color
        
    }

    //--------------------------------------------------------

}

